libvirt.go

```diff
@@ -316,20 +316,29 @@ func (d *Driver) Start() error {
 	// They wont start immediately
 	time.Sleep(5 * time.Second)
 
-	for i := 0; i < 90; i++ {
-		time.Sleep(time.Second)
+	for i := 0; i < 40; i++ {
 		ip, err := d.GetIP()
 		if err != nil {
 			return fmt.Errorf("%v: getting ip during machine start", err)
 		}
+
+		if ip == "" {
+			log.Debugf("Waiting for machine to come up %d/%d", i, 40)
+			time.Sleep(3 * time.Second)
+			continue
+		}
+
 		if ip != "" {
-			// Add a second to let things settle
-			time.Sleep(time.Second)
-			return nil
+			log.Infof("Found IP for machine: %s", ip)
+			d.IPAddress = ip
```

Comment:
ah ok this also adds some caching of the IP address in the Driver structure. Are you sure we don't need to keep the 'sleep' though?

Reply:
@cfergeau once we get the IP we don't need any sleep and return it to driver.

Reply:
I don't know, there was a comment implying it's desirable to wait a little bit more after getting the IP. I'm worried by not waiting anymore, we'll be introducing some races ?

Reply:
Alright if that make you bit uncomfortable, I will add that sleep :)